packages/components/src/util/clickable.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import classNames from "classnames/bind";
+import styles from "./clickable.module.css";
+
+const cx = classNames.bind(styles);
+
+export type ClickableProps<IComponent extends React.ElementType> = {
+  /**
+   * The style of the element.
+   */
+  variant: "flat" | "outline" | "minimal" | "link";
+  /**
+   * The color of the element.
+   */
+  color: "alert" | "brand" | "neutral" | "success" | "warning";
+  /**
+   * The size of the element.
+   */
+  size: "small" | "medium" | "large";
+  className: string;
+  /**
+   * A hidden prop for visual testing
+   */
+  state?: "inactive" | "hover" | "focus" | "disabled";
+} & React.ComponentProps<IComponent>;
+
+function Clickable<IComponent extends React.ElementType>({
+  as,
+  children,
+  color,
+  size,
+  state,
+  variant,
+  ...rest
+}: ClickableProps<IComponent>) {
+  const Component = as;
+  return (
+    <Component
+      className={cx(
+        `button`,
```

Comment:
would a normal link (rendered & styled as link) also go through this component do you think?

Reply:
yes

Reply:
Though this may be a reason to think of a name other than button - like `interactive` or `clickable` like it currently is. I like that clickable encompasses more than just button, but it feels too broad/meaningless honestly.